From fe57843466809053b702631e133cc30cfc6de55e Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Fri, 11 Nov 2022 19:16:23 +0300
Subject: [PATCH] add a new step to sign files with sign_impl command

Packages requiring signing will need to specify a sign_list
and sign_verify in APKBUILD files.
---
 abuild.in | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/abuild.in b/abuild.in
index 973f603..f76abae 100644
--- a/abuild.in
+++ b/abuild.in
@@ -796,6 +796,28 @@ build() {
 	:
 }
 
+sign() {
+	[ -z "$sign_list" ] && return 0
+
+	if ! command -v 'sign_impl' >/dev/null; then
+		warning "Please provide 'sign_impl' command/function to sign 'sign_list' (or sign manually)"
+		return 0
+	fi
+
+	local dir="${subpkgdir:-$pkgdir}"
+	local f
+	local cert=""
+
+	for f in $sign_list; do
+		local c; for c in $sign_verify; do
+			cert="$srcdir/$c"
+			[ -f "$cert" ] || die "Failed to find sign_verify file: $cert"
+
+			sign_impl "$dir"/$f "$cert"
+		done
+	done
+}
+
 # generate a simple tar.gz package of pkgdir
 targz() {
 	cd "$pkgdir" || return 1
@@ -1442,6 +1464,7 @@ prepare_package() {
 	[ ! -d "${subpkgdir:-$pkgdir}" ] && die "Missing ${subpkgname:+sub}pkgdir for ${subpkgname:-$pkgname}"
 	msg "Preparing ${subpkgname:+sub}package ${subpkgname:-$pkgname}..."
 	stripbin
+	sign
 	prepare_metafiles
 	prepare_trace_rpaths
 	prepare_symlinks
-- 
2.53.0

